Expose API via useImperativeHandle.

isofarro opened the pull request:
Following on from the suggestion in #14 , reimplemented my changes using a forwardRef and useImperativeHandle.

Implemented a `move()` method in the API that takes a long algebraic move (e.g. **e2e4**), and the method implementation splits it up into Chessground's preferred from-Key and to-Key.


Sample code demonstrating it's use:

```typescript
import { useEffect, useRef } from "react"
import { Config } from 'chessground/config'
import Chessground, { ApiRef, LongAlgebraicMove } from "./chessground/Chessground"

// these styles must be imported somewhere
import "chessground/assets/chessground.base.css";
import "chessground/assets/chessground.brown.css";
import "chessground/assets/chessground.cburnett.css";

const CONFIG: Config = {
  fen: "rnbq1rk1/1p3pbp/p2p1np1/2pP4/P3P3/2NB4/1P2NPPP/R1BQ1RK1 b - a3 0 10",
  lastMove: ["a2", "a4"],
  turnColor: "black",
  movable: {
    free: false
  }
}
  
const MOVES = "d8c7 e2g3 b8d7 h2h3 c5c4 d3c2 a8b8 c1e3 d7c5 a4a5 c8d7 d1d2 b7b5".split(" ")

export const BasicMovesView = () => {
  const apiRef = useRef<ApiRef | null>(null)

  useEffect(() => {
    const interval = setInterval(() => {
      const move: LongAlgebraicMove = MOVES.shift() as LongAlgebraicMove
      if (move) {
        apiRef.current!.move(move)
      } else {
        clearInterval(interval)
      }
    }, 2000)  
    return () => clearInterval(interval);
  })

  return (
    <Chessground width={640} height={640} config={CONFIG} ref={apiRef} />
  )
}
```

nowszy94 commented:
Would be nice to have api usage example in README.md

nowszy94 commented on src/index.tsx:
```diff
@@ -1,44 +1,66 @@
-    api?.set(config);
-  }, [api, config]);
+    useImperativeHandle(apiRef, () => {
+      return publicApi;
```
What about exposing the whole chessground api instead of creating custom object?

isofarro replied:
Sure, we can do that. Done now.

Also re-exported `Config`, `Api` and `Key` from the chessground API to reduce the clutter of imports in code using this React wrapper. Does that sound sensible?

isofarro commented:
> Would be nice to have api usage example in README.md

I've added the example demo-game component from this PR description to the README, and added `ref` to the props documentation.

nowszy94 commented on README.md:
````diff
@@ -40,3 +42,51 @@ import "chessground/assets/chessground.cburnett.css";
 ReactDOM.render(<Chessground />, document.getElementById("root"));
 ```
+
+## Example: showing the moves of a game
````
i really like this example!

nowszy94 commented on src/index.tsx:
```diff
@@ -1,44 +1,55 @@
+    useEffect(() => {
+      if (divRef.current && !api) {
+        const chessgroundApi = ChessgroundApi(divRef.current, {
+          animation: { enabled: true, duration: 200 },
```
lets remove it

isofarro replied:
Removed.

nowszy94 commented on src/index.tsx:
```diff
@@ -1,44 +1,55 @@
+          ...config,
+        });
+        setApi(chessgroundApi);
+      } else if (divRef.current && api) {
```
is it needed? It looks like it can only be fired after line 34 completes and after that the config is already provided

isofarro replied:
I gave it a quick test, and you're right, it doesn't look to be needed. Removed.

nowszy94 reviewed (approved):
Looks fantastic!
